continuum/scenarios/transformation_incremental.py

```diff
@@ -0,0 +1,68 @@
+from typing import Callable, List
+
+import numpy as np
+
+from continuum.task_set import TaskSet
+from continuum.datasets import _ContinuumDataset
+from continuum.scenarios import InstanceIncremental
+
+from torchvision import transforms
+
+
+class TransformationIncremental(InstanceIncremental):
+    """Continual Loader, generating datasets for the consecutive tasks.
+
+    Scenario: Every task contains the same data with different transformation
+    It is a cheap way to create instance incremental scenarios.
+    Classic transformation incremental scenarios are "permutations" and "rotations"
+
+    :param cl_dataset: A continual dataset.
+    :param nb_tasks: Number of tasks in the continuum.
+    :param incremental_transformations: list of transformations to apply to specific tasks
+    :param base_transformations: List of transformation to apply to all tasks.
+    """
+
+    def __init__(
+            self,
+            cl_dataset: _ContinuumDataset,
+            nb_tasks: int,
+            incremental_transformations: List[List[Callable]],
+            base_transformations: List[Callable] = None
+    ):
+        super(TransformationIncremental, self).__init__(
+            cl_dataset=cl_dataset,
+            nb_tasks=nb_tasks,
+            base_transformations=base_transformations
+        )
+        if incremental_transformations is None:
+            raise ValueError("For this scenario a list transformation should be set")
+
+        if nb_tasks != len(incremental_transformations):
+            raise ValueError("The number of tasks is not equal to the number of transformation")
+
+        self.inc_trsf = incremental_transformations
+        self._nb_tasks = self._setup(nb_tasks)
+
+    def get_task_index(self, nb_tasks, y):
+        # all tasks have all labels, only the transformation change
```

Comment:
Can we have an option to use either 'shared label space' (the case here) and the opposite case?

This paper https://arxiv.org/abs/2006.14769 has well defined this distinction in its section 2. and table 1.

Reply:
'shared label space' means that all tasks use the same labels ?

Reply:
Yes.

Task 1 is rot10°, task 2 rot20°, but the labels don't change.

Reply:
[patch_pr29 ebee5c8]